Daniel-Bennett777/Buildo-API
# new job_request

**To Test** checkout to main and git fetch this branch to test.

-  in the `job_requests.py` file be sure to check the content including the `create_job_request` and the `accept_contractor` decorators. 
- check `urls.py` file for the added job_requests registered router

- to Test functionality of the job_request go to postman and change the method to post method. Now enter this url into the top url : `http://localhost:8000/job_requests/create_job_request` then authorize into the headers with a contractors account : `d74b97fbe905134520bb236b0016703f50380dcf`      -- Now you can add in the request body `{
  "contractor": 1,  
  "customer": 2,    
  "work_order": 1,  
  "request_status": 1,  
  "contractor_cellphone": "123-456-7890"
}`

Be sure you get a 201 ok. 

**To test the accept_contractor**

-sign in as a customer account in postman. 
-no request body needed. into this url ;` http://localhost:8000/job_requests/1/accept_contractor` make sure its POST method. Be sure you get a 200 status code response. This dynamically accepted the job to the customer. 

----this also deleted the old accept job functionality that was within workorders. check to be sure def accept_job is gone from work_orders.py
